site/content/api/Tenant-API.md

```diff
@@ -0,0 +1,290 @@
+
+# Tenant Format
+
+Most of the operations of the Tenant API allow or require the inclusion of tenant data in the payload of the
+request or response messages of the operation. Such payload is carried in the body of the corresponding AMQP 
+messages as part of a single *AMQP Value* section.
+
+The tenant data is carried in the payload as a UTF-8 encoded string representation of a single JSON object. It is an error to include payload that is not of this type.
+
+The table below provides an overview of the standard members defined for the JSON object:
+
+| Name                     | Mandatory | Type       | Default Value | Description |
+| :------------------------| :-------: | :--------- | :------------ | :---------- |
+| *tenant-id*              | *yes*     | *string*   |               | The ID of the tenant. |
+| *enabled*                | *no*      | *boolean*  | `true`        | If set to `false` the tenant is currently disabled. Protocol adapters MUST NOT allow devices of a disabled tenant to connect and MUST NOT accept data published by such devices.
+| *adapters*   | *yes*     | *array*    |               | A list of configuration options valid for certain adapters only. See [Adapter Configuration Format]({{< relref "#adapter-configuration-format" >}}) for details. **NB** This array must contain at least one element - an empty array is handled as an error. **NB** Only a single entry per *type* is allowed. If multiple entries for the same type are present it is handled as an error.|
+
+Additionally to the specified properties the JSON object MAY contain an arbitrary number of members with arbitrary names which can be of a scalar or a complex type. 
+This allows for future *well-known* additions and also allows *clients* to add further information which might be e.g. relevant just for certain adapters only.
+
+### Adapter Configuration Format
+
+The table below contains the properties which are used to configure a *Hono protocol adapter*:
+
+| Name                               | Mandatory | Type       | Default Value | Description |
+| :--------------------------------- | :-------: | :--------- | :------------ | :---------- |
+| *type*                       | *yes*     | *string*   | `null`        | The type of the adapter which this configuration belongs to |
+| *enabled*                          | *no*      | *string*   | `true`        | If set to false the tenant is not allowed to receive / send data utilizing the given adapter. |
+| *device-authentication-required*   | *no*      | *string*   | `true`        | If set to false, devices are not required to authenticate with the adapter before sending / receiving data. |
+
+The adapter configuration of a tenant should be evaluated by the Hono Protocol Adapters.
```

Comment:
I would like to rephrase this sentence using the usual MAY, SHOULD, MUST notation in order to make the expectations regarding the adapters' behavior more explicit. What about
`Protocol Adapters SHOULD use the configuration properties set for a tenant when interacting with devices of that tenant, e.g. in order to make authorization decisions or to limit message rates per tenant etc.`

Reply:
I am not sure, though, how we should handle the *device-authentication-required* flag. In reality, we have no means to enforce this with a custom adapter, i.e. it will always be up to the protocol adapter to actually authenticate a device or not. However, we can limit the impact of it by restricting a protocol adapter's authority to get registration assertions to one or more tenants only.

Reply:
I took your phrase from the 1st comment. Furthermore I agree, we can't enforce what an adapter is doing with the information of the tenant api. It is the responsibility of the implementor of a certain adapter to use that tenant information in a proper way. This is some outline we should definitely add to the future multi tenancy concept page.